lib/spack/llnl/util/filesystem.py

```diff
@@ -1097,16 +1097,26 @@ def visit_directory_tree(root, visitor, rel_path='', depth=0):
     for f in dir_entries:
         if sys.version_info >= (3, 5, 0):
             rel_child = os.path.join(rel_path, f.name)
-            islink, isdir = f.is_symlink(), f.is_dir()
+            islink = f.is_symlink()
+            # Windows issue with reading symlinked archive
+            # results in [WinError 1921]
+            try:
+                isdir = f.is_dir()
+            except OSError:
+                isdir = False
+
         else:
             rel_child = os.path.join(rel_path, f)
             lexists, islink, isdir = lexists_islink_isdir(os.path.join(dir, f))
             if not lexists:
                 continue
 
-        if not isdir:
-            # Handle files
+        # handle symlinked files
```

Comment:
I think this comment needs to be edited: this first `if` is specifically for files that are *not* symlinks.